src/main/java/course_project/sea_battle/SeaBattleApp.java

```diff
@@ -0,0 +1,9 @@
+package course_project.sea_battle;
+
+import course_project.sea_battle.controller.impl.GameControllerImpl;
+
+public class SeaBattleApp {
```

Comment:
Opt: The field is quite narrow, i think it would look better with more space between cells by width
![image](https://user-images.githubusercontent.com/58872509/134136786-52412ebe-bf2e-4eaa-80b7-50242fad8030.png)

Reply:
The game seems not working. I see you implemented PvP, but I play only for 1 Player. I don't see field for 2nd, don't see his name anywhere. Don't know how to shoot for 2nd. 
And please introduce automatic ship placement, testing will be much easier

Reply:
Good! Nice architecture, SOLID is followed. Code is pretty readable. Good Java Core and Java libraries usage. 

To make better: interface could be refactored. Also check correct working of the game.

Reply:
Also nice approach to testing, test cases are readable

Reply:
Good, automatic ship placement! I found some minor issues with the interface: 
1) When miss, there's no message to press enter to finish the turn. Actually, I didn't see this message anywhere... Confusing
2) In case i hit a ship, i need to press enter 2 times to be able to shoot again
3) No error message in case I already shot this cell
![image](https://user-images.githubusercontent.com/58872509/134474624-47363b8a-6b01-4739-a488-2402b98bb992.png)
4) No message in case of destroyed ship 
![image](https://user-images.githubusercontent.com/58872509/134474815-1c7607f9-a49a-43ed-89f7-778bd4e5156f.png)
5) And please make field representation wider :) 

The app now works correctly, I didn't find any bugs. And from the code perspective, the app looks great! 

To make better: improve interface:) 
The homework is approved